File: frontend/types/student.ts
export interface Student {
  id: string;
  roll_number: string;
  name: string;
  email?: string;
  phone?: string;
  class_id: string;
  semester: number;
  section: string;
  is_active: boolean;
  created_at?: string;
}

export interface StudentCreate {
  roll_number: string;
  name: string;
  email?: string;
  phone?: string;
  class_id: string;
  semester: number;
  section: string;
}

export interface StudentUpdate {
  roll_number?: string;
  name?: string;
  email?: string;
  phone?: string;
  class_id?: string;
  semester?: number;
  section?: string;
  is_active?: boolean;
}

File: backend/app/schemas/student.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class StudentCreate(BaseModel):
    roll_number: str
    name: str
    email: Optional[str] = ""
    phone: Optional[str] = ""
    class_id: str
    semester: int
    section: str = "A"


class StudentUpdate(BaseModel):
    roll_number: Optional[str] = None
    name: Optional[str] = None
    email: Optional[str] = None
    phone: Optional[str] = None
    class_id: Optional[str] = None
    semester: Optional[int] = None
    section: Optional[str] = None
    is_active: Optional[bool] = None


class StudentResponse(BaseModel):
    id: str
    roll_number: str
    name: str
    email: Optional[str] = ""
    phone: Optional[str] = ""
    class_id: str
    semester: int
    section: str
    is_active: bool
    created_at: Optional[datetime] = None
